From 306a9832b9c80fb96bdad34b9d6d732bcfccdcc8 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sun, 30 Jan 2022 13:43:29 +0100
Subject: [PATCH 035/172] Add tilde('~') to gopher regex for chkURL

The tilde ('~') is valid character in gopher URLs and is often used to denote a
user's home directory.

---
 main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/main.c
+++ b/main.c
@@ -5073,7 +5073,7 @@ chkURLBuffer(Buffer *buf)
 	"https?://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*[a-zA-Z0-9_/=\\-]",
 	"file:/[a-zA-Z0-9:%\\-\\./=_\\+@#,\\$;]*",
 #ifdef USE_GOPHER
-	"gopher://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./_]*",
+	"gopher://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./_~]*",
 #endif				/* USE_GOPHER */
 	"ftp://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./=_+@#,\\$]*[a-zA-Z0-9_/]",
 #ifdef USE_NNTP
